### PR TITLE
1077 update dtypes to better support scalars

### PR DESCRIPTION
**Pull request by hokiegeek2**
This PR is intended to do the following:

1. Streamline dtypes
2. Broaden scope of dtypes int_scalars
3. Add unit test cases to encompass new int_scalars scope

### CODE REVIEW

File: tests/extrema_test.py [hunk 1]
```diff
@@ -86,9 +86,12 @@ def test_error_handling(self):
         with self.assertRaises(TypeError):
             ak.mink(list(range(0, 10)), 1)
 
-        with self.assertRaises(TypeError):
-            ak.mink(testArray, "1")
-
+        with self.assertRaises(TypeError) as cm:
+            ak.mink(testArray, '1')
+        self.assertEqual(('type of argument "k" must be one of (int, int8, int16, int32, ' +
```
Comment (Ethan-DeBandi99) [inline tests/extrema_test.py:91]: Just marking this in one spot. I know @pierce314159 and I have had issues in the past when we check the specific error message. I think in general we have been assuming the correct error being raised is enough.

**Review by Ethan-DeBandi99 — CHANGES_REQUESTED** [review]
Requesting changes because it looks like we need to rerun black (and maybe isort). Nothing major just some of the formatting looks modified from what was done last week. Can we run `black` and `isort` on the modified files? I marked on of the tests, but we should make a decision moving forward about checking the full error message in a test. We have experienced issues with this in the past and have been staying away from it.

File: tests/extrema_test.py (hunk 1 above)
Comment (pierce314159) [reply tests/extrema_test.py:91]: Yeah I'm of the opinion that it's better not the check the exact error message. We've seen strangeness before where users have seen test failures when there's nothing wrong

There was a case where where we were checking for:
`'type of argument "bins" must be one of (int, int64, uint64); got str instead'`
and users were getting errors cause there's said:
`'type of argument "bins" must be one of (int, np.int64, uint64); got str instead'`
https://github.com/Bears-R-Us/arkouda/issues/1105

So I think just verifying it's a `TypeError` should be enough

**Comment by hokiegeek2** [comment]
> Requesting changes because it looks like we need to rerun black (and maybe isort). Nothing major just some of the formatting looks modified from what was done last week. Can we run `black` and `isort` on the modified files? I marked on of the tests, but we should make a decision moving forward about checking the full error message in a test. We have experienced issues with this in the past and have been staying away from it.

@Ethan-DeBandi99 yeah, I found this tricky. I ran autopep8 on the arkouda directory and that was a disaster. I ran pep8 on only the files flagged by flake8 and that worked. In terms of black and sort, what is the overall workflow for all this? pep8 -> black -> sort -> flake8? Please confirm and I'll get all the changed files updated accordingly. :+1:

**Comment by pierce314159** [comment]
> > Requesting changes because it looks like we need to rerun black (and maybe isort). Nothing major just some of the formatting looks modified from what was done last week. Can we run `black` and `isort` on the modified files? I marked on of the tests, but we should make a decision moving forward about checking the full error message in a test. We have experienced issues with this in the past and have been staying away from it.
> 
> @Ethan-DeBandi99 yeah, I found this tricky. I ran autopep8 on the arkouda directory and that was a disaster. I ran pep8 on only the files flagged by flake8 and that worked. In terms of black and isort, what is the overall workflow for all this? pep8 -> black -> isort -> flake8? Please confirm and I'll get all the changed files updated accordingly. 👍

I've found that `isort` -> `black` -> `flake8` to be a good workflow for this. I've not used the `autopep8`, but `black` and `flake8` will cover everything u need

**Comment by hokiegeek2** [comment]
> > > Requesting changes because it looks like we need to rerun black (and maybe isort). Nothing major just some of the formatting looks modified from what was done last week. Can we run `black` and `isort` on the modified files? I marked on of the tests, but we should make a decision moving forward about checking the full error message in a test. We have experienced issues with this in the past and have been staying away from it.
> > 
> > 
> > @Ethan-DeBandi99 yeah, I found this tricky. I ran autopep8 on the arkouda directory and that was a disaster. I ran pep8 on only the files flagged by flake8 and that worked. In terms of black and isort, what is the overall workflow for all this? pep8 -> black -> isort -> flake8? Please confirm and I'll get all the changed files updated accordingly. 👍
> 
> I've found that `isort` -> `black` -> `flake8` to be a good workflow for this. I've not used the `autopep8`, but `black` and `flake8` will cover everything u need. And you're right it's not very clear, I'm going to update `CONTRIBUTING.md` to give more info on this soon

gotcha, okay, looks like isort did nothing, so I guess it's on board w/ pep8. I ran black and that tidied up things:

image

I'll update the others w/ isort -> black and submit the changes -> thanks dude! 👍

**Comment by hokiegeek2** [comment]
@Ethan-DeBandi99 @pierce314159 updated formatting on 4 files with isort -> black flow, checked w/ flake8

File: tests/extrema_test.py (hunk 1 above)
Comment (hokiegeek2) [reply tests/extrema_test.py]: @Ethan-DeBandi99 @hokiegeek2 removed error message checking

**Review by Ethan-DeBandi99 — APPROVED** [review]
Looks like you ran black with the default line length. Not a problem, just for future reference, we have been running it with `--line-length 105`. I should have mentioned that before, but it will not hurt anything in the current state. Also, it looks like the tests are back to `'`, which again shouldn't break things, but I believe black with flip flop that. Overall, the changes look good.

**Review by reuster986 — APPROVED** [review]

**Comment by hokiegeek2** [comment]
@pierce314159 @jeichert60 please weigh in on this PR when y'all get a chance. Once you've both approved, I will fix the timeclass.py conflict in prep for merge into master branch

@mhmerrill FYI

**Review by joshmarshall1 — APPROVED** [review]

**Review by jeichert60 — APPROVED** [review]

**Comment by hokiegeek2** [comment]
@pierce314159 fixed merge conflict and flake8 error, ready for final review and merge

**Review by pierce314159 — APPROVED** [review]
Looks good!

**Pull request merged**